File: TSDW/Esercitazioni/itinere3/client_order_s/src/main/java/com/example/client_order_s/controllers/ClientController.java
package com.example.client_order_s.controllers;

import org.springframework.stereotype.*;
import org.springframework.web.bind.annotation.*;
import org.springframework.ui.*;

import com.example.client_order_s.models.Client;
import com.example.client_order_s.repositories.ClientRepository;

@Controller
public class ClientController {
    
    private ClientRepository repo;

    public ClientController(ClientRepository repo){
        this.repo=repo;
    }

    @GetMapping("/")
    public String home(){
        return "index";
    }

    @GetMapping("/clients")
    public String index(Model model){
        model.addAttribute("clients", repo.findAll());
        return "clients/index";
    }

    @GetMapping("/clients/create")
    public String crete(){
        return "clients/create";
    }

    @GetMapping("/clients/edit")
    String edit(Long id, Model model){
        model.addAttribute("client", repo.findById(id).orElse(null));
        return "/clients/edit";
    }

    @GetMapping("/clients/show")
    String show(Long id, Model model){
        model.addAttribute("client", repo.findById(id).orElse(null));
        return "/clients/show";
    }

    @PostMapping("/clients/store")
    public String store(Client client){
        repo.save(client);
        return "redirect:/clients";
    }

    @PostMapping("/clients/update")
    public String update(Client client){
        repo.save(client);
        return "redirect:/clients";
    }

    @PostMapping("/clients/delete")
    public String delete(Long id){
        repo.deleteById(id);
        return "redirect:/clients";
    }
    
}

File: TSDW/Esercitazioni/itinere3/client_order_l/app/Http/Controllers/ClientController.php
<?php

namespace App\Http\Controllers;

use App\Models\Client;
use Illuminate\Http\Request;

class ClientController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view('clients.index', ["clients"=>Client::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view('clients.create');
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Client::create($request->all());
        return redirect()->route("clients.index");
    }

    /**
     * Display the specified resource.
     */
    public function show(Client $client)
    {
        return view('clients.show', ["client"=>$client]);
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Client $client)
    {
        return view("clients.edit", ["client"=>$client]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Client $client)
    {
        $client->update($request->all());
        return redirect()->route("clients.index");
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Client $client)
    {
        $client->delete();
        return redirect()->route('clients.index');
    }
}
